### PR TITLE
ID-239 Upgrade Sam Java Version to 17

### PR DESCRIPTION
**Pull request by tlangs**
Ticket: https://broadworkbench.atlassian.net/browse/ID-239
Upgrading Sam to Java 17 is something we should do anyways, but it also has the side-effect of using a docker image with an arm64 build, so Sam can run on M1 laptops! We still need to get rid of OpenDJ to actually boot Sam, but this would need to happen anyways for the sbt file-change watching!

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://broadinstitute.github.io/dsp-appsec-security-risk-assessment/) and attached the result to the JIRA ticket

### CODE REVIEW

**Comment by tlangs** [comment]
jenkins retest

**Comment by tlangs** [comment]
jenkins retest

**Comment by tlangs** [comment]
jenkins retest

**Review by gpolumbo-broad — APPROVED** [review]

**Comment by tlangs** [comment]
jenkins retest

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -46,10 +46,10 @@ jobs:
         run: |
           sh docker/run-opendj.sh start
 
-      - name: Set up JDK 11
+      - name: Set up JDK 17
         uses: actions/setup-java@v1
```
Comment (dexamundsen) [inline .github/workflows/build.yml:50]: Could we use v3

    - name: Set up JDK 17
      uses: actions/setup-java@v3
      with:
        distribution: 'temurin'
        java-version: 17

File: README.md [hunk 2]
```diff
@@ -138,7 +138,7 @@ Note that Sam does not actually launch workflows create VMs but appears to in th
 ## Development
 ### Required Tooling:
 #### Java:
-Make sure you have Java JDK 11 installed. Instructions for our recommended package can be found [here](https://adoptopenjdk.net/)
+Make sure you have Java JDK 17 installed. Instructions for our recommended package can be found [here](https://adoptopenjdk.net/)
```
Comment (dexamundsen) [inline README.md]: Minor - 
change link to https://adoptium.net/

File: docker/build.sh [hunk 3]
```diff
@@ -101,10 +101,10 @@ function make_jar()
...
+    docker run --rm --link postgres:postgres --link $OPENDJ:opendj -e DIRECTORY_URL=$DIRECTORY_URL -e GIT_MODEL_HASH=$GIT_MODEL_HASH -e DIRECTORY_PASSWORD=$DIRECTORY_PASSWORD -v $PWD:/working -v jar-cache:/root/.ivy -v jar-cache:/root/.ivy2 hseeberger/scala-sbt:eclipse-temurin-17.0.2_1.6.2_2.13.8 /working/docker/init_schema.sh /working
     docker restart $OPENDJ
     sleep 40
-    docker run --rm --link postgres:postgres --link $OPENDJ:opendj -e DIRECTORY_URL=$DIRECTORY_URL -e GIT_MODEL_HASH=$GIT_MODEL_HASH -e DIRECTORY_PASSWORD=$DIRECTORY_PASSWORD -v $PWD:/working -v jar-cache:/root/.ivy -v jar-cache:/root/.ivy2 broadinstitute/scala-baseimage:jdk11-2.13.5-1.4.7 /working/docker/install.sh /working
+    docker run --rm --link postgres:postgres --link $OPENDJ:opendj -e DIRECTORY_URL=$DIRECTORY_URL -e GIT_MODEL_HASH=$GIT_MODEL_HASH -e DIRECTORY_PASSWORD=$DIRECTORY_PASSWORD -v $PWD:/working -v jar-cache:/root/.ivy -v jar-cache:/root/.ivy2 hseeberger/scala-sbt:eclipse-temurin-17.0.2_1.6.2_2.13.8 /working/docker/install.sh /working
     EXIT_CODE=$?
     set -e # Turn error detection back on for the rest of the script
```
Comment (dexamundsen) [inline docker/build.sh:110]: How about also adding the target platform for docker images built at lines 128 & 132?

imageNameAndTag="$DOCKERHUB_REGISTRY:${HASH_TAG}"
targetPlatform="--platform=linux/amd64"
docker build -t $imageNameAndTag $targetPlatform  ... other args

build script executed on M1 macs (arm64) also produce standard amd64 images that cane be used widely. 
(This helped me while working on a different repo)

Comment (tlangs) [reply docker/build.sh:110]: Unfortunately, this breaks Jenkins's ability to build the image.
image

**Comment by tlangs** [comment]
jenkins retest

File: docker/build.sh (hunk 3 above)
Comment (dexamundsen) [reply docker/build.sh:110]: noted, thanks for checking

**Review by dexamundsen — APPROVED** [review]

**Pull request merged**